src/lavinmq/clustering/replicator.cr

```diff
@@ -9,6 +9,8 @@ module LavinMQ
       abstract def append(path : String, obj)
       abstract def delete_file(path : String, wg : WaitGroup)
       abstract def followers : Array(Follower)
+      abstract def syncing_followers : Array(Follower)
+      abstract def all_followers : Array(Follower)
```

Comment:
Maybe not related to this PR, but I wonder if it's possible to change the return type of these to `Iterator(Follower)` to be able to return empty iterators instead of empty arrays in `NoopServer`? It would save us some allocations.